### PR TITLE
Include a configurable test timeout

### PR DESCRIPTION
**Pull request by sschneid**
in seconds, default: `1800` (30m)

### CODE REVIEW

**Comment by sschneid** [comment]
@mesosphere-ci retest this please

**Comment by sschneid** [comment]
@mesosphere-ci retest this please

**Comment by sschneid** [comment]
@mesosphere-ci retest this please

File: shakedown/cli/main.py [hunk 1]
```diff
@@ -313,7 +314,7 @@ def pytest_sessionfinish(self, session, exitstatus):
                 for output in shakedown.stdout:
                     echo(output)
 
-    opts = ['-q', '--tb=no']
+    opts = ['-q', '--tb=no', '--timeout='+str(args['timeout'])]
```
Comment (kensipe) [inline shakedown/cli/main.py]: should this be '--timeout={}'.format(args['timeout'])'?

Comment (kensipe) [inline shakedown/cli/main.py]: does  `--timeout` tie to the pytest-timeout?

Comment (sschneid) [inline shakedown/cli/main.py]: Yes, it is used by the `pytest-timeout` plugin -- and I'll update to change to using `format`, one sec...

**Comment by kensipe** [comment]
oh... and how about supporting this in `~/.shakedown` ?

**Comment by kensipe** [comment]
tested working!

**Comment by sschneid** [comment]
> how about supporting this in `~/.shakedown` ?

Yup, the way the configfile is handled, this is automatically available in `.shakedown` (via something like `timeout = 1800`).

**Comment by sschneid** [comment]
Test failures unrelated to this PR, will be fixed in subsequent PR.

**Pull request merged**